src/Razor/src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs

```diff
@@ -76,6 +76,13 @@ internal class Pass : IntermediateNodePassBase, IRazorDirectiveClassifierPass
 
             protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
             {
+                if (documentNode.DocumentKind != RazorPageDocumentClassifierPass.RazorPageDocumentKind &&
```

Comment:
We do this check for other mvc-only directives but looks like we missed this one.